apps/emqx_eviction_agent/src/emqx_eviction_agent.erl

```diff
@@ -364,22 +379,23 @@ do_evict_sessions(Nodes, ChannelStream) ->
                     %% taken over by a reconnecting client, so it is already lost.
                     %%
                     %% Therefore, it is safe to just discard the session here.
-                    discard_session_channel(ClientId)
+                    discard_session_channel(ClientId, ChanPid)
             end
         end,
         ChannelStream
     ).
 
-is_session_evictable(ClientId) ->
-    emqx_cm_registry:lookup_channels(ClientId) =/= [].
+is_session_evictable(ClientId, ChanPid) ->
+    emqx_cm_registry:lookup_channels(ClientId) =/= [] andalso
+        is_process_alive(ChanPid).
```

Comment:
Q: is `ChanPid` guaranteed to always be a node-local pid?  Otherwise, `is_process_alive` will crash.

Reply:
Yes, emqx_cm registers local channels only